README.md

```diff
@@ -47,24 +51,26 @@ Ensure you have `docker` and `docker-compose` installed in your system.
 
 Currently, documentation from OpenROAD and OpenROAD-flow-scripts is chunked recursively and embedded into FAISS Vector Databases.  
 
-Upon receiving a query, relevant documents are retrieved and reranked, using [LangChain's Ensemble Retriever](https://python.langchain.com/v0.1/docs/modules/data_connection/retrievers/ensemble/).
+Documents are first retrieved from the vectorstore using a hybrid retriever, combining vector and semantic search methods. These retrieved documents undergo reranking using a cross-encoder reranker model.
 
-Retrieved documents are then sent to the LLM as input context, for generating a response.
+Then top-n documents from the reranked set are then sent to the LLM as input context, for generating a response.
 
 ```mermaid
 flowchart LR
-    id1([Query]) --> id2([OR Docs])
-    id1([Query]) --> id3([ORFS Docs])
+    id0([Query]) --> id1
 
-    id2([OR Docs]) --> id4([Reranking]) 
-    id3([ORFS Docs]) --> id4([Reranking])
+    id1([Vectorstore]) --- id2([Semantic Retriever])
+    id1([Vectorstore]) --- id3([MMR Retriever])
+    id1([Vectorstore]) --- id4([BM25 Retriever])
 
-    id4([Reranking]) --> id5([LLM])
+    id2([Semantic Retriever]) -- Retrieved Docs ---> id5([Reranking]) 
+    id3([MMR Retriever]) -- Retrieved Docs ---> id5([Reranking])
+    id4([BM25 Retriever]) -- Retrieved Docs ---> id5([Reranking])
+
+    id5([Reranking]) -- top-n docs --> id6([LLM])
  
 ```
 
-By default,
-
 The backend will then be hosted at [http://0.0.0.0:8000](http://0.0.0.0:8000). 
 
 Open [http://0.0.0.0:8000/docs](http://0.0.0.0:8000/docs) for the API docs.
```

Comment:
Insert some docs on how to use the Ruff to format and check code. 

Also are you planning it as a pre-commit / github CI? Or separate PR.

Reply:
A `mypy` type checker would be good as well if time permits.